src/prefect/orion/database/migrations/versions/postgresql/2023_01_26_152801_0a1250a5aa25_expand_work_queue_table.py

```diff
@@ -0,0 +1,365 @@
+            ["work_pool_id", "priority"],
+            unique=False,
+        )
+
+    # Create default agent work pool and associate all existing queues with it
+    connection = op.get_bind()
+
+    connection.execute(
+        sa.text(
+            "INSERT INTO work_pool (name, type) VALUES ('default-agent-pool', 'prefect-agent')"
+        )
+    )
+
+    default_pool_id = connection.execute(
+        sa.text("SELECT id FROM work_pool WHERE name = 'default-agent-pool'")
+    ).fetchone()[0]
+
+    default_queue = connection.execute(
+        sa.text("SELECT id FROM work_queue WHERE name = 'default'")
+    ).fetchone()
+
+    if not default_queue:
+        connection.execute(
+            sa.text(
+                f"INSERT INTO work_queue (name, work_pool_id) VALUES ('default', :default_pool_id)"
+            ).params({"default_pool_id": default_pool_id}),
+        )
+
+    connection.execute(
+        sa.text(
+            "UPDATE work_queue SET work_pool_id = :default_pool_id WHERE work_pool_id IS NULL"
+        ).params({"default_pool_id": default_pool_id}),
+    )
+
+    default_queue_id = connection.execute(
+        sa.text(
+            "SELECT id FROM work_queue WHERE name = 'default' and work_pool_id = :default_pool_id"
+        ).params({"default_pool_id": default_pool_id}),
+    ).fetchone()[0]
+
+    connection.execute(
+        sa.text(
+            "UPDATE work_pool SET default_queue_id = :default_queue_id WHERE id = :default_pool_id"
+        ).params(
+            {"default_pool_id": default_pool_id, "default_queue_id": default_queue_id}
+        ),
+    )
+
+    # Set priority on all queues and update flow runs and deployments
+    queue_rows = connection.execute(
+        sa.text(
+            "SELECT id, name FROM work_queue WHERE work_pool_id = :default_pool_id"
+        ).params({"default_pool_id": default_pool_id}),
+    ).fetchall()
+
+    with op.get_context().autocommit_block():
+        for enumeration, row in enumerate(queue_rows):
+            connection.execute(
+                sa.text(
+                    "UPDATE work_queue SET priority = :priority WHERE id = :id"
+                ).params({"priority": enumeration + 1, "id": row[0]}),
+            )
+
+            batch_size = 250
+
+            while True:
+                result = connection.execute(
+                    sa.text(
+                        """
+                        UPDATE flow_run 
+                        SET work_queue_id=:id 
+                        WHERE flow_run.id in (
+                            SELECT id 
+                            FROM flow_run 
+                            WHERE flow_run.work_queue_id IS NULL and flow_run.work_queue_name=:name 
+                            LIMIT :batch_size
+                        )
+                        """
+                    ).params({"id": row[0], "name": row[1], "batch_size": batch_size}),
+                )
+                if result.rowcount <= 0:
```

Comment:
I've found using `result.rowcount < batch_size` to be a more robust break condition; I've run into some problems using `<= 0` before

Reply:
Good call! I implemented this in https://github.com/PrefectHQ/prefect/pull/8327